tests/unit/s2n_certificate_signatures_test.c

```diff
@@ -0,0 +1,175 @@
+            EXPECT_TRUE(out);
+
+            BIO_free(certBio);
+            X509_free(cert);
+        }
+
+        /* Certificate signature algorithm is not in test certificate signature preferences list */
+        {
+            EXPECT_SUCCESS(s2n_read_test_pem(S2N_RSA_PSS_2048_SHA256_LEAF_CERT, (char *)cert_file, S2N_MAX_TEST_PEM_SIZE));
+            certLen = strlen((const char*)cert_file);
+
+            /* Read the test certificates into an Openssl X509 struct */
+            certBio = BIO_new(BIO_s_mem());
+            BIO_write(certBio, cert_file, certLen);
+            cert = PEM_read_bio_X509(certBio, NULL, NULL, NULL);
+            S2N_ERROR_IF(cert == NULL, S2N_ERR_DECODE_CERTIFICATE);
+
+            EXPECT_OK(s2n_is_certificate_sig_scheme_supported(conn, cert, &test_certificate_signature_preferences, &out));
+            EXPECT_FALSE(out);
+
+            BIO_free(certBio);
+            X509_free(cert);
+        }
+
+        /* Certificate signature algorithm is in the test certificate signature preferences list but signature is SHA-1
+         * and TLS 1.3 has been negotiated.
+         */
+        {
+            conn->actual_protocol_version = S2N_TLS13;
+            EXPECT_SUCCESS(s2n_read_test_pem(S2N_RSA_2048_PKCS1_CERT_CHAIN, (char *)cert_file, S2N_MAX_TEST_PEM_SIZE));
+            certLen = strlen((const char*)cert_file);
+
+            /* Read the test certificates into an Openssl X509 struct */
+            certBio = BIO_new(BIO_s_mem());
+            BIO_write(certBio, cert_file, certLen);
+            cert = PEM_read_bio_X509(certBio, NULL, NULL, NULL);
+            S2N_ERROR_IF(cert == NULL, S2N_ERR_DECODE_CERTIFICATE);
+
+            EXPECT_OK(s2n_is_certificate_sig_scheme_supported(conn, cert, &test_certificate_signature_preferences, &out));
+            EXPECT_FALSE(out);
+
+            BIO_free(certBio);
+            X509_free(cert);
+        }
+
+        /* Certificate signature algorithm is in the test certificate signature preferences list and signature is SHA-1
+         * and TLS 1.2 has been negotiated.
+         */
+        {
+            conn->actual_protocol_version = S2N_TLS12;
+            EXPECT_SUCCESS(s2n_read_test_pem(S2N_RSA_2048_PKCS1_CERT_CHAIN, (char *)cert_file, S2N_MAX_TEST_PEM_SIZE));
+            certLen = strlen((const char*)cert_file);
+
+            /* Read the test certificates into an Openssl X509 struct */
+            certBio = BIO_new(BIO_s_mem());
+            BIO_write(certBio, cert_file, certLen);
+            cert = PEM_read_bio_X509(certBio, NULL, NULL, NULL);
+            S2N_ERROR_IF(cert == NULL, S2N_ERR_DECODE_CERTIFICATE);
+
+            EXPECT_OK(s2n_is_certificate_sig_scheme_supported(conn, cert, &test_certificate_signature_preferences, &out));
+            EXPECT_TRUE(out);
+
+            BIO_free(certBio);
+            X509_free(cert);
+        }
+
+        /* RSA PSS certificates can be parsed */
```

Comment:
Isn't what we needed to test a signature made with an RSA PSS certificate, not the RSA PSS certificate itself? We want to test the case where we expect the signature nid to be NID_rsassaPss?

Reply:
This test checks the S2N_RSA_PSS_2048_SHA256_LEAF_CERT which has been signed by the S2N_RSA_PSS_2048_SHA256_CA_CERT with an rsa pss algorithm.

Reply:
The comment is inaccurate, and it's not obvious from the test itself that S2N_RSA_PSS_2048_SHA256_LEAF_CERT has an RSA-PSS signature. I'd make the comment clearer about what you're testing.